src/main/java/org/opensearch/knn/index/query/FilteredKNNScorer.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query;
+
+import java.io.IOException;
+
+public interface FilteredKNNScorer {
```

Comment:
> We create class called as FilteredKNNScorer(exactly same as FilteredFloatKNNScorer) and NestedFieldFilteredKNNScorer.
The class NestedFieldFilteredKNNScorer extends FilteredKNNScorer and can override any functions it requires.

1. FilteredFloatKNNScorer and NestedFieldFilteredFloatKNNScorer do not have child-parent relationship. They are different. 
2. FilteredKNNScorer is interface instead of abstract class or parent class because there is no class level attributes needed.
3. I have this naming pattern to support Byte type in the future.

> we should avoid that duplication too.
I can remove the duplication.

>  we should remove the prefix of Scorer
Please check https://github.com/apache/lucene/blob/7b8aece125aabff2823626d5b939abf4747f63a7/lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenFloatKnnVectorQuery.java#L162 where Scorer suffix is used with the exactly same functionality.